service.js

```diff
@@ -48,21 +48,38 @@ function getUrlMetadata(url) {
 }
 
 
-const app = express();
-app.use(bodyParser.json()); // for parsing application/json
-
-app.post('/', function(req, res) {
-  const promises = req.body.urls.map((url) => {
-    return getUrlMetadata(url);
-  });
+const server = new Hapi.Server();
+server.connection({
+  port: 7001
+});
 
-  Promise.all(promises).then((urlsData) => {
-    res.json({
-      error: '',
-      urls: buildObj(urlsData.map((urlData) => [urlData.url, urlData]))
-    });
-  });
+server.route({
+  method: 'POST',
+  path: '/',
+  config: {
+    validate: {
+      payload: {
+        urls: Joi.array().required()
+      }
+    }
+  },
+  handler: (req, reply) => {
+    const promises = req.payload.urls.map((url) => getUrlMetadata(url));
+
+    Promise.all(promises)
```

Comment:
Unrelated to this garbage PR, but it feels like this should have some `.catch((err) => {...})` flow here in case one of the promises is rejected for some reason.